protocol/dubbo3/dubbo3_invoker.go

```diff
@@ -0,0 +1,142 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dubbo3
+
+import (
+	"context"
+	"reflect"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+)
+
+import (
+	hessian2 "github.com/apache/dubbo-go-hessian2"
+)
+
+import (
+	"github.com/apache/dubbo-go/common"
+	"github.com/apache/dubbo-go/common/constant"
+	"github.com/apache/dubbo-go/common/logger"
+	"github.com/apache/dubbo-go/config"
+	"github.com/apache/dubbo-go/protocol"
+	invocation_impl "github.com/apache/dubbo-go/protocol/invocation"
+	"github.com/apache/dubbo-go/remoting/dubbo3"
+)
+
+// Dubbo3Invoker is implement of protocol.Invoker. A dubboInvoker refer to one service and ip.
+type Dubbo3Invoker struct {
+	protocol.BaseInvoker
+	// the exchange layer, it is focus on network communication.
+	client   *dubbo3.TripleClient
+	quitOnce sync.Once
+	// timeout for service(interface) level.
+	timeout time.Duration
+	// Used to record the number of requests. -1 represent this DubboInvoker is destroyed
+	reqNum int64
+}
+
+// NewDubbo3Invoker constructor
+func NewDubbo3Invoker(url *common.URL) (*Dubbo3Invoker, error) {
+	requestTimeout := config.GetConsumerConfig().RequestTimeout
+	requestTimeoutStr := url.GetParam(constant.TIMEOUT_KEY, config.GetConsumerConfig().Request_Timeout)
+	if t, err := time.ParseDuration(requestTimeoutStr); err == nil {
+		requestTimeout = t
+	}
+
+	client, err := dubbo3.NewTripleClient(url)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Dubbo3Invoker{
+		BaseInvoker: *protocol.NewBaseInvoker(url),
+		client:      client,
+		reqNum:      0,
+		timeout:     requestTimeout,
+	}, nil
+}
+
+// Invoke call remoting.
+func (di *Dubbo3Invoker) Invoke(ctx context.Context, invocation protocol.Invocation) protocol.Result {
+	var (
+		result protocol.RPCResult
+	)
+
+	var in []reflect.Value
+	in = append(in, reflect.ValueOf(ctx))
```

Comment:
using 'in := append(in, reflect.ValueOf(ctx))' instead

Reply:
r u sure?
I guess change 
`var in []reflect.Value `
to
`in := make(reflect.Value, 0, 16)`
may be better?

Reply:
change it as ur idae